src/lib/auth/auth.ts

```diff
@@ -0,0 +1,139 @@
+/* Facade for keycloak */
+import Keycloak, { KeycloakInitOptions, KeycloakInstance } from 'keycloak-js';
+
+export interface ResourceRole {
+  resource: string;
+  resourceId: string;
+  role: string;
+}
+
+export interface OrganizationRole {
+  orgNr: string;
+  role: string;
+}
+
+export interface AuthConfiguration {
+  oidcIssuer: string;
+  clientId: string;
+  redirectUri: string;
+  logoutRedirectUri: string;
+}
+
+export interface AuthInitOptions {
+  loginRequired: boolean;
+}
+
+export interface Auth {
+  init(options: AuthInitOptions): Promise<boolean>;
+
+  login(): Promise<void>;
+
+  logout(): Promise<void>;
+
+  isAuthenticated(): boolean;
+
+  getUser(): User;
+
+  getToken(): Promise<string>;
+
+  getAuthorizationHeader(): Promise<string>;
+
+  getAuthorities(): string;
+
+  getResourceRoles(): ResourceRole[];
+
+  hasResourceRole(r: ResourceRole): boolean;
+
+  hasOrganizationRole(organizationRole: OrganizationRole): boolean;
+
+  hasOrganizationReadPermission(orgNr: string): boolean;
+
+  hasOrganizationWritePermission(orgNr: string): boolean;
+
+  hasOrganizationAdminPermission(orgNr: string): boolean;
+
+  hasSystemAdminPermission(): boolean;
+}
+
+export interface User {
+  username: string;
+  name: string;
+}
+
+export const nullUser = { username: '', name: '' };
+
+const extractUser = tokenParsed => ({ username: tokenParsed.user_ame, name: tokenParsed.name });
+
+export class Auth implements Auth {
```

Comment:
Perhaps rename auth interface to AuthInterface to avoid name collision

Reply:
Vi har spesual lintregel som forbyr bruk av I-prefix. 

https://stackoverflow.com/questions/43055682/relationship-between-a-typescript-class-and-an-interface-with-the-same-name/43056090

Nå forstå jeg egentlig, hva har skjedde

vi må tenke litt om det:
https://stackoverflow.com/a/41967120/3251100

Reply:
jeg tror vi er nå enig, at hvis implementasjon og interface er identiske, da er det bedre å bruke implementasjon som interface.